app/lib/ui/views/welcome/welcome_screen.dart

```diff
@@ -1,38 +1,30 @@
-import 'package:app/core/view_models/welcome_view_model.dart';
-import 'package:app/ui/views/home/home_screen.dart';
-import 'package:app/ui/views/navigation_drawer.dart';
-import 'package:app/ui/views/signin/sign_in_screen.dart';
+import 'package:app/core/view_models/welcome_screen_view_model.dart';
 import 'package:app/ui/widgets/utility/template_base_view_model.dart';
 import 'package:flutter/material.dart';
 
 class WelcomeScreen extends StatelessWidget {
   final String title;
 
-  const WelcomeScreen({Key key, this.title}) : super(key: key);
+  WelcomeScreen({Key key, this.title}) : super(key: key);
   static const route = '/';
 
   @override
   Widget build(BuildContext context) {
-    return TemplateBaseViewModel<WelcomeViewModel>(
+    return TemplateBaseViewModel<WelcomeScreenViewModel>(
       builder: (context, model, child) => Scaffold(
         appBar: AppBar(),
-        drawer: NavigationDrawer(),
         body: Center(
           child: Column(
             mainAxisAlignment: MainAxisAlignment.center,
             children: [
               Text('Welcome!'),
               RaisedButton(
-                onPressed: () {
-                  Navigator.of(context).pushNamed(SignInScreen.route);
-                },
+                onPressed: model.navigateToSignInScreen,
                 child: Text('Sign In'),
               ),
               // TODO: Remove eventually, useful for testing now
               RaisedButton(
-                onPressed: () {
-                  Navigator.of(context).pushNamed(HomeScreen.route);
-                },
+                onPressed: model.navigateToHomeScreen,
```

Comment:
I think you forgot to add `()` at the end?

Reply:
I did not! This is an uncommon way of specifying the function to call.

This is the same as `() => model.navigateToHomeScreen()`
> So, rather than make an anonymous function to call a function, I just say call this function `model.navigateToHomeScreen`. 

TLDR; functions can be pointed to like variables, adding `()` calls it, but `() => {}` declares a function.